core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -199,6 +199,8 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
   private val quotaCallback = clientQuotaCallback.getOrElse(new DefaultQuotaCallback)
   private val staticConfigClientIdQuota = Quota.upperBound(config.quotaDefault.toDouble)
   private val clientQuotaType = QuotaType.toClientQuotaType(quotaType)
+  private val quotaViolationLastLoggedTimestamp = new ConcurrentHashMap[String, Long]();
```

Comment:
We need to have a record expiration mechanism here to prevent quotaViolationLastLoggedTimestamp growing to big. I think for Kafka Rest, it is using a new random client id each time, so quotaViolationLastLoggedTimestamp would keep growing with new incoming random client id that violates quota.

Reply:
Thanks for taking the review! I actually had the mechanism as part of the PR. Essentially once the request for the same sensor is not violating the quota anymore, we remove the record from the map.